hr_job_category/models/hr.py

```diff
@@ -23,16 +23,32 @@ class HRJob(models.Model):
 class HRContract(models.Model):
     _inherit = "hr.contract"
 
-    def _tag_employees(self, job_id):
-        if job_id:
-            job = self.env["hr.job"].browse(job_id)
-            self.mapped("employee_id").write(
-                {"category_ids": [(6, 0, job.category_ids.ids)]}
-            )
-        else:
-            for contract in self:
-                categories = contract.job_id and contract.job_id.category_ids.ids or []
-                contract.employee_id.write({"category_ids": [(6, 0, categories)]})
+    def _remove_tags(self, employee_id=None, job_id=None):
+        # TODO write tags only once
+        if not employee_id or not job_id:
+            return
```

Comment:
Easy to test that condition for better test coverage. Or maybe this method does not need to return a value